src/platform/packages/shared/response-ops/rule_form/src/rule_definition/rule_dashboards.tsx

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import React, { useState, useEffect, useCallback } from 'react';
+import {
+  EuiComboBox,
+  EuiSplitPanel,
+  EuiFlexItem,
+  EuiSpacer,
+  EuiTitle,
+  EuiComboBoxOptionOption,
+} from '@elastic/eui';
+import { RuleTypeParams } from '@kbn/alerting-types';
+import { dashboardServiceProvider, type DashboardItem } from './dashboard_service';
+import { useRuleFormState, useRuleFormDispatch } from '../hooks';
+import { ALERT_LINK_DASHBOARDS_TITLE } from '../translations';
+import { type RuleDashboardsPlugins } from '../types';
+
+export interface RuleDashboardsPluginsProps {
+  plugins: RuleDashboardsPlugins;
+}
+
+interface DashboardOption {
+  value: string;
+  label: string;
+}
+
+interface RuleTypeParamsWithDashboards extends RuleTypeParams {
+  dashboards?: Array<{ id: string }>;
+}
+
+export const RuleDashboards = ({ plugins }: RuleDashboardsPluginsProps) => {
+  const { featureFlags, dashboard: dashboardService } = plugins;
+  const { formData } = useRuleFormState();
+  const dispatch = useRuleFormDispatch();
+  // const dashboardsFormData = formData.dashboards;
```

Comment:
TODO: I plan to rewrite this part and save linked dashboards in a separate dashboards property [here](https://github.com/elastic/kibana/blob/main/src/platform/packages/shared/response-ops/rule_form/src/types.ts#L43)

```
export interface RuleFormData<Params extends RuleTypeParams = RuleTypeParams> {
  .....
  flapping?: Rule<Params>['flapping'];
  dashboards?: Array<{ id: string }>;
}
```